### PR TITLE
Bug 1872465: Collect missing logs at scale

### PR DESCRIPTION
**Pull request by jcantrill**
### Description
This PR mitigates log collection at scale  by introducing a dual container collector:
* Fluentbit to watch and collect logs, pushing them to a normalizer to assist with log rotation
* Fluentd acting as a normalizer at the edge

/cc @RH-ematysek @syedriko 
/assign @alanconway 

/hold

### Links
Ref:

https://bugzilla.redhat.com/show_bug.cgi?id=1872465
[LOG-679](https://issues.redhat.com/browse/LOG-679)

This PR will require productization of fluentbit before merge

### CODE REVIEW

**Comment by jcantrill** [comment]
replaces https://github.com/openshift/cluster-logging-operator/pull/762

File: pkg/k8shandler/collection.go [hunk 1]
```diff
@@ -54,36 +54,33 @@ func (clusterRequest *ClusterLoggingRequest) CreateOrUpdateCollection(proxyConfi
...
 		}
 
 		if err = clusterRequest.reconcileFluentdServiceMonitor(); err != nil {
-			return
+			log.V(3).Error(err, "Error reconciling fluentd servicemonitor")
 		}
 
 		if err = clusterRequest.createOrUpdateFluentdPrometheusRule(); err != nil {
-			return
+			log.V(3).Error(err, "Error reconciling fluentd prometheusrules")
```
Comment (alanconway) [inline pkg/k8shandler/collection.go:65]:
```suggestion
			log.V(3).Error(err, "Error reconciling fluentd prometheus rules")
```

**Review by alanconway — APPROVED** [review]
Do you have before/after benchmarks for the performance improvement?
Probably needs more work but we should get this in so we can progress it.

/lgtm

**Comment by jcantrill** [comment]
Closing in favor of addressing as a larger scale solution https://issues.redhat.com/browse/LOG-1179

**Pull request closed without merging**